OneCommunityGlobal/HighestGoodNetworkApp
# Cherrypick Eiki's Task Edit Suggestions Part 2

This PR is the second part of [Eiki's Task Edit Suggestions PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/497/commits), due to the large scale of changes and commits, we decided to cherrypick all his changes and split into couple new PRs, which would be more readable and understandable. You can also see the first cherrypick PR [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/536)

**Mainly changes explained:** 🔮 
1. delete all the mock data on the frontend, change to populate data from backend
2. create the ability to do tasks suggestions for volunteer/core team user
3. implement the reject function on the taskEditSuggestion

**How to test:** 🔍 

1. check on the [related backend PR](https://github.com/OneCommunityGlobal/HGNRest/pull/186)
2. check on this PR as frontend
4. log as volunteer/core team account
5. Dashboard -> Tasks and Timelogs -> Tasks tab -> click one of the tasks -> click the "suggest" button -> make some changes on the popup table -> **click "update" button**
<img width="942" alt="image" src="https://user-images.githubusercontent.com/5071040/200686673-7c3b3482-4bcd-4b20-85a9-cae2f11b6eb1.png">

6. log out and log in as a admin/owner user
7. go to http://localhost:3000/taskeditsuggestions
8. click on the row on taskEditSuggestions page to open the suggested changes, click the "Reject" button
<img width="862" alt="image" src="https://user-images.githubusercontent.com/5071040/200687257-13f7864b-531e-4e4f-b5d2-fcb803669634.png">

**Note: below functionalities will be implemented in coming PRs** 💡 
1. tasks suggest function only be accessed by manager/mentor users, instead, volunteer/core team users can only view tasks
2. taskEditSuggestion page will only be accessed by admin/owner users
